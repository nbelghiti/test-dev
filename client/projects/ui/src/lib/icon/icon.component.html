<i class="c-icon__glyph {{ size && 'c-icon__glyph--' + size }} {{ type && 'c-icon__glyph--' + type }} fas fa-{{ name }}"></i>
