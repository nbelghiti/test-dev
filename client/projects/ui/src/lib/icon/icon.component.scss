// =============================================================================
// :: Imports
// =============================================================================
@import "global";

// =============================================================================
// :: Icon
// =============================================================================
:host {
	display: inline-flex;
	justify-content: center;
	align-items: center;

	font-size: inherit;
}

// =============================================================================
// :: Elements
// =============================================================================
.c-icon__glyph {
	line-height: inherit;

	cursor: pointer;
}

// =============================================================================
// :: Modifiers
// =============================================================================
.c-icon__glyph--large {
	font-size: 2rem;
}

.c-icon__glyph--small {
	font-size: 1.4rem;
}

.c-icon__glyph--tiny {
	font-size: 1.2rem;
}

.c-icon__glyph--success {
	color: $ui-color-success-dark;
}

.c-icon__glyph--error {
	color: $ui-color-error-dark;
}

.c-icon__glyph--warning {
	color: $ui-color-warning-dark;
}

.c-icon__glyph--info {
	color: $ui-color-info-dark;
}
