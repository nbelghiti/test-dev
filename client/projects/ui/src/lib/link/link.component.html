<button
	*ngIf="hasClick; else link"
	type="button"
	class="c-link {{ type }}"
	(click)="clicked.emit()"
>
	<div *ngIf="iconBefore" class="c-link__icon c-link__icon--before">
		<ui-icon [name]="iconBefore"></ui-icon>
	</div>

	<span class="c-link__text">{{label}}</span>

	<div *ngIf="iconAfter" class="c-link__icon c-link__icon--after">
		<ui-icon [name]="iconAfter"></ui-icon>
	</div>
</button>

<ng-template #link>
	<a
		class="c-link {{ type }}"
		[routerLink]="url"
	>
		<div *ngIf="iconBefore" class="c-link__icon c-link__icon--before">
			<ui-icon [name]="iconBefore"></ui-icon>
		</div>

		<span class="c-link__text">{{label}}</span>

		<div *ngIf="iconAfter" class="c-link__icon c-link__icon--after">
			<ui-icon [name]="iconAfter"></ui-icon>
		</div>
	</a>
</ng-template>
