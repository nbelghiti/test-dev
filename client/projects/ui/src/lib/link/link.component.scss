// =============================================================================
// :: Imports
// =============================================================================
@import "global";

// =============================================================================
// :: Link
// =============================================================================
.c-link {
	display: inline-flex;
	align-items: center;

	text-decoration: none;

	color: $primary-color;

	transition: color $animation-base;

	&:hover {
		color: darken($primary-color, 8%);
	}
}

// =============================================================================
// :: Elements
// =============================================================================
.c-link__text {
	text-decoration: underline;
	font-size: $font-size-1x-small;
}

// =============================================================================
// :: Modifiers
// =============================================================================
.c-link__icon--before {
	margin-right: 0.8rem;
}

.c-link__icon--after {
	margin-left: 0.8rem;
}

.c-link--info {
	color: $ui-color-info-dark;

	.c-link__text {
		font-size: 1.6rem;
	}

	&:hover {
		color: darken($ui-color-info-dark, 10%);
	}
}
