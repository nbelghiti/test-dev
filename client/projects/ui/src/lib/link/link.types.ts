export enum LinkType {
	primary = '',
	info = 'c-link--info',
}
