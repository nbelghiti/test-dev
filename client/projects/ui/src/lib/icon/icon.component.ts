import { Component, HostBinding, Input } from '@angular/core';

import { IconName, IconSize, IconType } from './icon.types';

@Component({
	selector: 'ui-icon',
	templateUrl: './icon.component.html',
	styleUrls: ['./icon.component.scss']
})

export class IconComponent {
	@Input() public name: IconName;
	@Input() public type: IconType = null;
	@Input() public size: IconSize = null;

	@HostBinding('class.c-icon') public className = true;
}
