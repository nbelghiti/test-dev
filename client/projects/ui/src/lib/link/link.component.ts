import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';

import { LinkType } from './link.types';


@Component({
	selector: 'ui-link',
	templateUrl: './link.component.html',
	styleUrls: ['./link.component.scss']
})

export class LinkComponent implements OnInit {
	@Input() public type?: LinkType;
	@Input() public label: string;
	@Input() public url: string;
	@Input() public iconBefore: string;
	@Input() public iconAfter: string;
	@Output() public clicked: EventEmitter<Event> = new EventEmitter<Event>();

	public hasClick: boolean = false;

	public ngOnInit(): void {
		this.hasClick = this.clicked.observers.length > 0;
	}
}
