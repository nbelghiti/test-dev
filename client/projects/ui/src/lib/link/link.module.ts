import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';

import { IconModule } from '../icon/icon.module';

import { LinkComponent } from './link.component';

@NgModule({
	imports: [
		CommonModule,
		IconModule,
		RouterModule,
	],
	declarations: [
		LinkComponent,
	],
	exports: [
		LinkComponent,
	],
})
export class LinkModule {}
