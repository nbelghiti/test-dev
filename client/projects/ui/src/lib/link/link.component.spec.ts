import { ComponentFixture, TestBed, async } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';

import { IconComponent } from '../icon/icon.component';

import { LinkComponent } from './link.component';

describe('LinkComponent', () => {
	let component: LinkComponent;
	let fixture: ComponentFixture<LinkComponent>;

	beforeEach(async(() => {
		TestBed.configureTestingModule({
			imports: [ RouterTestingModule ],
			declarations: [
				LinkComponent,
				IconComponent
			]
		})
		.compileComponents();
	}));

	beforeEach(() => {
		fixture = TestBed.createComponent(LinkComponent);
		component = fixture.componentInstance;
		fixture.detectChanges();
	});

	it('should create', () => {
		expect(component).toBeTruthy();
	});

	it('Should render correctly', () => {
		expect(fixture).toMatchSnapshot();
	});
});
