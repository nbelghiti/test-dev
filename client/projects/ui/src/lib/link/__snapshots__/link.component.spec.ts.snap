// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkComponent Should render correctly 1`] = `
<ui-link
  clicked={[Function EventEmitter]}
  hasClick="false"
>
  <a
    class="c-link "
    href="/"
  >
    
    <span
      class="c-link__text"
    >
      
    </span>
    
  </a>
</ui-link>
`;
